Restaurant page features

helciodev opened the pull request:
The features of this project are, user can:

1. page change dynamically
2. Javascript modules.

helciodev committed:
Create package.json file

helciodev committed:
Add .github/workflows .eslintrc.json, .hintrc and .styllintrc.json files

helciodev committed:
Add node_modules to g.gitignore file
install webpack and webpack-cli as dev dependencies

helciodev committed:
Add script in package.json file and add entry and outuput in webpack config file;

helciodev committed:
Add navigation as an adjacentHTML to the container,
add module folder and home.js , menu.js and contact.js files

helciodev committed:
Append home as child for the container create function addClasses inside home.js file

helciodev committed:
Create function appendChildren inside home.js file

helciodev committed:
Add classes to menu elements

helciodev committed:
Add content for menu elements

helciodev committed:
Appendchildren for home div

helciodev committed:
Add assets folder

helciodev committed:
Import functions from home module to menu module and create some HTML elements for the menu

helciodev committed:
Add classes for menu elements

helciodev committed:
Add content for dish and headingOneMenu elements

helciodev committed:
Delete rule from webpack.config.js file

helciodev committed:
Delete image hamburger.jpg
add dish3 and dish4
add images for dishes

helciodev committed:
Style menu elements

helciodev committed:
Fix eslint offense
create elements for contact and style it

helciodev committed:
Add logic to change content in the page

helciodev committed:
Initial dist subtree commit

helciodev committed:
Add logic to change content and delete dist folder from .gitignore file

helciodev committed:
Fix eslint offenses

helciodev committed:
Change assets folder directory

helciodev committed:
Update the README file

helciodev committed:
Update README.md

helciodev committed:
Update the README file

helciodev committed:
Merge branch 'restaurant-page-features' of github.com:helciodev/JS-restaurant-page into restaurant-page-features

helciodev committed:
Update the README file

afanimartin commented on README.md:
```diff
@@ -1,2 +1,88 @@
+## Getting Started
+
+
+### Installation instructions
```
- Add instructions for installing packages, building the app locally and running

afanimartin commented on package.json:
```diff
@@ -0,0 +1,24 @@
+  "version": "1.0.0",
+  "description": "Restaurant page with the pages build entirely with JavaScript and Webpack for modularization",
+  "main": "index.js",
+  "scripts": {
+    "build": "webpack --watch"
+  },
```
- Add missing commands
```json
"scripts": {
    "dev": "webpack --mode development",
    "build": "webpack --mode production",
    "start": "webpack-dev-server --mode development --open" // optional
  },
```

afanimartin commented on modules/home.js:
```diff
@@ -0,0 +1,43 @@
+function addClasses(elem, classes) {
```
- Use ES6 syntax for creating functions

afanimartin commented on modules/home.js:
```diff
@@ -0,0 +1,43 @@
+  classes.split(' ').forEach((clas) => elem.classList.add(clas));
+}
+
+function appendChildren(node, ...children) {
```
- Use ES6 synatx

afanimartin commented on modules/contact.js:
```diff
@@ -0,0 +1,11 @@
+import { appendChildren, addClasses } from './home';
+
+const contactContainer = document.createElement('div');
+const headingOneContact = document.createElement('h1');
+
+addClasses(headingOneContact, 'text-3xl gray-500 font-bold opacity-100');
+addClasses(contactContainer, 'mx-auto w-1/2 bg-white opacity-90 px-5 py-3 text-center');
+
+headingOneContact.textContent = '222-333-444';
+appendChildren(contactContainer, headingOneContact);
```
- Implement this module using ES6 function syntax and instead export that function

afanimartin commented on modules/menu.js:
```diff
@@ -0,0 +1,55 @@
+headingOneMenu.textContent = "Gary's menu";
+
+appendChildren(menuContainer, headingOneMenu, dish, dish2, dish3, dish4);
+export default menuContainer;
```
- Implement this module using ES6 syntax for functions and export that function

helciodev replied:
thanks for the very descriptive code review!

afanimartin reviewed (changes requested):
## Changes Required 🙏🏿 🙏🏿 🙏🏿 
Nice work on this project. I like the design of your app.
### Well done 🌵 🌵 🌵 
- use of modules
- necessary packages installed

### Changes to make 🙅🏿 🙅🏿 🙅🏿 
- update readme with instructions for getting started locally. Check my comments.
- use Es6 syntax for creating functions

### Martin Afani
#### Slack: whiz25

helciodev committed:
Upadate installation instructions and add missing npm commands

helciodev committed:
Update the README file
rewiret addClasses function using ES6 syntax
Install webpack-deve-server and add commands in webpack.config.js to run the server in ./dist folder

helciodev committed:
Rewrite function appendChildren using ES6 syntax
export modules as arrow functions

oloomoses reviewed (approved):
## Congratulations, Project Aproved! :tada: :muscle: 
**Hello @helciodev ,
Well done on making the changes suggested by @afanimartin. You have passed all the requirements for this project :100:** 

Happy Codding and best wishes in your next project

_slack: Oloo Moses_
